pkg/registry/blobs.go

```diff
@@ -20,11 +20,15 @@ import (
 	"encoding/hex"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"math/rand"
 	"net/http"
 	"path"
 	"strings"
 	"sync"
+
+	"github.com/google/go-containerregistry/pkg/name"
```

Comment:
I've been trying to keep this from having any dependencies outside the stdlib.

Reply:
Too late: https://github.com/google/go-containerregistry/blob/0dfbb568512eae093cfeaefe620aabec76a6681b/pkg/registry/manifest.go#L31-L32

Reply:
Generally, I agree with this goal for *implementations* of the interfaces, but it seems wasteful to reimplement validation that exists in the `name` and similar packages.

Reply:
https://github.com/google/go-containerregistry/pull/798#discussion_r515407375

Reply:
Jon beat me to it 🙈

Reply:
My comment stands.  It's foolish to reimplement the validation we have in `name` and other places that let us avoid dealing with stringly typed data, and I don't see the risk of a dependency cycle for these superficial things because they should absolutely NOT need a full-blown registry implementation to test interfaces.

I could absolutely get behind a `depcheck` style test that this never depend on `pkg/v1/remote` in particular.